hvac/utils.py

```diff
@@ -280,3 +282,31 @@ def _check_pem(arg):
     if not isinstance(param_argument, list) or not all(_check_pem(p) for p in param_argument):
         error_msg = 'unsupported {param} public key / certificate format, required type: PEM'
         raise exceptions.ParamValidationError(error_msg.format(param=param_name))
+
+
+def format_url(format_str, *args, **kwargs):
+    """Creates a URL using the specified format after escaping the provided arguments.
+
+    :param format_str: The URL containing replacement fields.
+    :type format_str: str
+    :param kwargs: Positional replacement field values.
+    :type kwargs: list
+    :param kwargs: Named replacement field values.
+    :type kwargs: dict
+    :return: The formatted URL path with escaped replacement fields.
+    :rtype: str
+    """
+
+    def maybe_quote(maybe_str):
+        if isinstance(maybe_str, six.string_types):
+            return six.moves.urllib.parse.quote(maybe_str)
+        else:
+            return maybe_str
```

Comment:
For maximum safety, can we cast `maybe_str` to a str instead of using this conditional branch? Presumably if we had some class that returns unsafe characters for use in a HTTP request when cast to a string, we still want to ensure that `__str__` return value is escaped?

Reply:
This sounds like a good idea, I'm going to need to do some thinking about how to handle this in a Python 2/3 compatible way.  (The biggest challenge will be Unicode strings.  Python 2's urllib throws an exception but Python 3 is perfectly happy with them.)

Reply:
I've reworked the URL quoting to _always_ cast to a Unicode type, UTF-8 encode it, then run it through urllib quoting.  This gives the same result on Python 2 and 3.  The only time this fails is if the passed object can't be converted to text, but that would have eventually resulted in failure anyways.